mtf-readings test fails on local-only diagnostic dirs (orphan slug check)

**Severity:** minor

## Expected

`npm test` passes on a clean checkout of `main`, including after running the MTF digitizer locally for an unaccepted lens slug.

## Actual

`src/data/mtf-readings.test.ts` > `docs/optical-specs directory-name invariant` > `every optical-specs directory matches a lens via toSlug` fails when local-only diagnostic directories exist under `docs/optical-specs/` for slugs that do not match a lens in `lenses.ts`.

Concrete repro: after running the digitizer for ``tokina-atx-m-11-18mm-f2-8-x-at-11mm`` / ``-at-18mm``, two diagnostic-only directories are created on disk. These are .gitignored per ADR-050 (`docs/optical-specs/*/diagnostic/`), so the contents are untracked — but the empty parent directories still trip the test:

```
AssertionError: optical-specs directories with no matching lens (slug drift?):
  tokina-atx-m-11-18mm-f2-8-x-at-11mm,
  tokina-atx-m-11-18mm-f2-8-x-at-18mm
```

## Reproduce

1. Run the MTF digitizer for any slug that does not exist as a lens in `lenses.ts` — e.g. a focal-length variant of an existing zoom (`<parent-slug>-at-11mm`)
2. Observe a directory created under `docs/optical-specs/<unmatched-slug>/diagnostic/`
3. Run `npm test`
4. Result: orphan test fails on the per-FL directories

## Environment

dev (local), main branch (clean tree). Repros independent of the eslint-plugin-unicorn v65 bump (#1153) — discovered while validating that PR locally.

## Root cause

`lensSpecDirs` in `mtf-readings.test.ts:265` uses `readdirSync` to list every directory under `docs/optical-specs/`. It does not distinguish tracked content from local-only artifacts. The check then enforces that every directory matches a lens slug, which makes the test fail on local-only digitizer scratch directories.

## Possible fixes

1. **Require a tracked anchor file.** Only include directories that contain at least one tracked file (e.g. `analysis.md`, `digitization-log.md`, or `specs-log.md`) — local-only diagnostic dirs would be skipped naturally.
2. **Use `git ls-tree`** to enumerate tracked directories only.
3. **Skip directories whose only content is the ignored `diagnostic/` subdir** — fragile, do not recommend.

(1) seems cleanest: matches the intent (the test is about authored content), no git command in the test runner, and aligns with the existing `digitization-log.md` anchor in the sibling `mtfReadings` coverage check at `mtf-readings.test.ts:276`.

## Notes

- Related: Tokina ATX-M 11-18 (parent slug `tokina-atx-m-11-18mm-f2-8-x`) is in `lenses.ts` as a zoom; the per-FL slugs are MTF reference-set / digitizer artifacts, not separate lenses.
- Local workaround: delete the orphan dirs to unblock `npm test`.
- This is the second class of orphan handled by this test — the first is `KNOWN_PENDING_LENS_ENTRY` at line 315 for surveyed-but-unmodeled lenses. A tracked-anchor-file filter would subsume both classes.
- Discovered while verifying #1153 (eslint-plugin-unicorn v65); does not block the bump.

--- a/src/data/mtf-readings.test.ts
+++ b/src/data/mtf-readings.test.ts
@@ -262,9 +262,18 @@ const opticalSpecsDir = resolve(process.cwd(), "docs/optical-specs");
 // Directories with a leading underscore are staging areas for
 // pre-collected materials (e.g. `_pending-mitakon-cine/`) and are
 // intentionally not lens slugs — exclude them from slug-shape checks.
+//
+// Directories whose only content is `diagnostic/` (ADR-050: per-stage
+// digitizer bundle, gitignored) are local-only artifacts from a
+// `mtfdigitizer diagnose` run on a slug that does not exist as a
+// lens. They appear on disk but never in git; excluding them keeps
+// `npm test` deterministic across machines (#1183).
 const lensSpecDirs: string[] = readdirSync(opticalSpecsDir).filter((entry) => {
   if (entry.startsWith("_")) return false;
-  return statSync(join(opticalSpecsDir, entry)).isDirectory();
+  const full = join(opticalSpecsDir, entry);
+  if (!statSync(full).isDirectory()) return false;
+  const trackedEntries = readdirSync(full).filter((e) => e !== "diagnostic");
+  return trackedEntries.length > 0;
 });
 
 describe("docs/optical-specs ↔ mtf-readings coverage", () => {

